src/HttpCache/EventListener/AddTagsListener.php

```diff
@@ -64,6 +70,33 @@ public function onKernelResponse(FilterResponseEvent $event)
             return;
         }
 
+        $resources = $this->removeDisabledResourcesFromCacheTags($resources);
+
         $response->headers->set('Cache-Tags', implode(',', $resources));
     }
+
+    private function removeDisabledResourcesFromCacheTags($resources)
+    {
+        $resourceCacheHeadersPerResourceClass = [];
+        foreach ($resources as $resource) {
+            if (3 !== \count(explode('/', $resource))) { // simple check if it's an item or collection resource
+                continue;
+            }
+            $resourceClass = \get_class($this->iriConverter->getItemFromIri($resource));
```

Comment:
The behat tests are failling because those classes could be Doctrine Proxies . You need to import and use the `ApiPlatform\Core\Util\ClassInfoTrait` and then modify this line like so


```suggestion
            $resourceClass =  $this->getObjectClass($this->iriConverter->getItemFromIri($resource));
```